test/integration_test/utils/marketplace.go

```diff
@@ -250,3 +317,39 @@ func GetCurrentCSV(name, namespace string) (string, error) {
 	}
 	return currentCSV, nil
 }
+
+// CreatePxVsphereSecret creates secret for PX to authenticate with vSphere
+func CreatePxVsphereSecret(namespace, pxVsphereUsername, pxVspherePassword string) error {
+	logrus.Infof("Creating secret [%s]", defaultPxVsphereSecretName)
+	name, _ := base64.StdEncoding.DecodeString(pxVsphereUsername)
+	password, _ := base64.StdEncoding.DecodeString(pxVspherePassword)
+
+	secret := core_v1.Secret{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      defaultPxVsphereSecretName,
+			Namespace: namespace,
+		},
+		Type: "Opaque",
+		Data: map[string][]byte{
+			"VSPHERE_USER":     name,
+			"VSPHERE_PASSWORD": password,
+		},
+	}
+
+	t := func() (interface{}, bool, error) {
+		if _, err := core.Instance().CreateSecret(&secret); err != nil {
+			if k8s_errors.IsAlreadyExists(err) {
+				logrus.Warnf("Secret [%s] already exists, reusing it.", secret.Name)
+				return nil, false, nil
+			}
+			return nil, true, fmt.Errorf("failed to create secret [%s], Err: %v", secret.Name, err)
+		}
+		return nil, false, nil
+	}
+
+	if _, err := task.DoRetryWithTimeout(t, getPxVsphereSecretTimeout, getPxVsphereSecretRetryInterval); err != nil {
```

Comment:
why keep retrying creating it?

Reply:
We have seen an issue where sometime connection causes the reject or fail some particular APIs, like creating secret and in this particular call we can call it again to make sure it does get created instead of the whole job failing due to some latency issues. I have copied this func from `spawn` we added this there for a reason as it helped the job not fail due to infra